util/metrics/kubectl_metrics.go

```diff
@@ -0,0 +1,360 @@
+)
+
+var (
+	clientCertRotationAgeGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "argocd_kubectl_client_cert_rotation_age_seconds",
+		Help: "Age of a certificate that has just been rotated",
+	}, []string{"initiator"})
+
+	requestLatencyHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "argocd_kubectl_request_latency_seconds",
+		Help:    "Kubectl request latency",
+		Buckets: []float64{0.01, 0.05, 0.10, 0.25, .5, 1},
+	}, []string{"initiator", "hostname", "verb"})
+
+	resolverLatencyHistogram = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "argocd_kubectl_resolver_latency_seconds",
+			Help:    "Kubectl resolver latency",
+			Buckets: []float64{0.01, 0.05, 0.10, 0.25, .5, 1},
+		},
+		[]string{"initiator", "hostname"},
+	)
+
+	requestSizeSummary = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name: "argocd_kubectl_request_size_bytes",
+			Help: "Size of kubectl requests",
+			Objectives: map[float64]float64{
+				0.5:  0.05,
+				0.9:  0.01,
+				0.95: 0.001,
+				0.99: 0.001,
+			},
+		},
+		[]string{"initiator", "hostname", "method"},
+	)
+
+	responseSizeSummary = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name: "argocd_kubectl_response_size_bytes",
+			Help: "Size of kubectl responses",
+			Objectives: map[float64]float64{
+				0.5:  0.05,
+				0.9:  0.01,
+				0.95: 0.001,
+				0.99: 0.001,
+			},
+		},
+		[]string{"initiator", "hostname", "method"},
+	)
+
+	rateLimiterLatencyHistogram = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "argocd_kubectl_rate_limiter_latency_seconds",
+			Help:    "Kubectl rate limiter latency",
+			Buckets: []float64{0.01, 0.05, 0.10, 0.25, .5, 1},
+		},
+		[]string{"initiator", "hostname", "verb"},
+	)
+
+	requestResultCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "argocd_kubectl_request_result_total",
+			Help: "Number of kubectl request results",
+		},
+		[]string{"initiator", "hostname", "method", "result"},
+	)
+
+	execPluginCallsCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "argocd_kubectl_exec_plugin_calls_total",
+			Help: "Number of kubectl exec plugin calls",
+		},
+		[]string{"initiator", "exit_code", "call_status"},
+	)
+
+	requestRetryCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "argocd_kubectl_request_retry_total",
+			Help: "Number of kubectl request retries",
+		},
+		[]string{"initiator", "hostname", "method", "code"},
+	)
+
+	transportCacheEntriesGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "argocd_kubectl_transport_cache_entries_total",
+			Help: "Number of kubectl transport cache entries",
+		},
+		[]string{"initiator"},
+	)
+
+	transportCreateCallsCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "argocd_kubectl_transport_create_calls_total",
+			Help: "Number of kubectl transport create calls",
+		},
+		[]string{"initiator", "result"},
```

Comment:
What is the value the "result" label will have?

Reply:
Added to docs